### PR TITLE
refactor(core): Make instance role clearer (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
https://n8nio.slack.com/archives/C069HS026UF/p1721899050522709?thread_ts=1721894117.050629&cid=C069HS026UF

### CODE REVIEW

File: packages/cli/src/config/schema.ts [hunk 1]
```diff
@@ -917,12 +917,13 @@ export const schema = {
 		},
 	},
 
+	instanceRole: {
```
Comment (netroy) [inline packages/cli/src/config/schema.ts:920]: why are we keeping this state in the config though? can this not be a property on one of the services instead?
we should stop using config to store data that user can't configure.

Comment (ivov) [reply packages/cli/src/config/schema.ts:920]: I wouldn't feel comfortable making that change without [thorough retesting](https://github.com/n8n-io/n8n/pull/8446).

Comment (ivov) [reply packages/cli/src/config/schema.ts:920]: This was likely done following what was done with `instanceType`. I think there's value in making `instanceRole` clearer until we decide to remove all state from the schema. Let me know what you prefer.

Comment (netroy) [reply packages/cli/src/config/schema.ts:920]: does this make sense? https://github.com/n8n-io/n8n/compare/make-instance-role-clearer...10188-proposal?expand=1

Comment (ivov) [reply packages/cli/src/config/schema.ts:920]: Thanks! In principle it does, what I'm nervous about is merging this without thorough testing.

**Review by netroy — APPROVED** [review]

File: packages/cli/src/config/schema.ts (hunk 1 above)
Comment (ivov) [reply packages/cli/src/config/schema.ts:920]: Let's bring it in. I'll do some spot checks testing tomorrow but I agree it looks safe.

Comment (netroy) [reply packages/cli/src/config/schema.ts:920]: https://github.com/n8n-io/n8n/pull/10242

**Comment by janober** [comment]
Got released with [`n8n@1.53.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.53.0)

**Pull request merged**